Xamarin.Forms.Core/VisualElement.cs

```diff
@@ -783,9 +784,26 @@ static void FlowDirectionChanged(BindableObject bindable, object oldValue, objec
 			self.NotifyFlowDirectionChanged();
 		}
 
+		static void OnIsEnabledPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			if (!(bindable is VisualElement element))
+			{
+				return;
+			}
+
+			var isEnabled = (bool)newValue;
+
+			VisualStateManager.GoToState(element, isEnabled 
```

Comment:
Please don't call the VisualStateManager directly. There should be a method such as "UpdateVisualState" that is called whenever the VisualElement thinks the visual state should be updated. This should be avirtual method so that developers can easily sub-class and override the visual state manager behaviour.